### ISSUE TITLE
Routing Errors with NextJS 13's App Directory

### ISSUE THREAD

**Issue by PixelPage-YT** [aralroca/next-translate#1101]
**What version of this package are you using?**
2.4.4

**What operating system, Node.js, and npm version?**
macOS 13.4.1 (22F82) // Node.js v18.16.0 // npm 9.5.1 // NextJS 13.4

**What i did**
I used the app router and added all my pages to /[lang].
I also added the path in my i18n.json:
```json
{
  "locales": ["en"],
  "defaultLocale": "en",
  "pages": {
    "/[lang]/(index)": ["index"], (I added all my files for / in the folder (index) and wasn't sure which one to use)
    "/[lang]": ["index"],
    "/[lang]/contact": ["contact"],
    "/[lang]/story": ["story"]
  }
}
```

**What happened?**
When i go to my page, the link can't be found (Error: 404).
When i go to /en, the page gets displayed but the translations do not get loaded.

**What did you expect to happen?**
After going to /, you are redirected to the correct locale with the correct translations loaded.

**Are you willing to submit a pull request to fix this bug?**
/

**Comment by aralroca** [comment]
@PixelPage-YT would you share a reproducible example? thank you so much!

**Comment by PixelPage-YT** [comment]
> @PixelPage-YT would you share a reproducible example? thank you so much!

hmm, i tried doing it but in my example everything worked fine. Probably an error with my specific app, i'm gonna try to find it.

**Comment by PixelPage-YT** [comment]
Okay, the problem still exists.
I upgraded all my packages, and then it worked, so I continued translating my navbar.
But then, it didn't work again, even when I removed `"*": ["layout"],` from my i18n.json. Very weird.

My config:
```
{
  "locales": ["en", "de"],
  "defaultLocale": "en",
  "pages": {
    "*": ["layout"],
    "/[lang]": ["index"],
    "/": ["index"],
    "/[lang]/contact": ["contact"],
    "/contact": ["contact"],
    "/[lang]/story": ["story"],
    "/story": ["story"]
  }
}
```

The correct namespaces are loaded (At least the console says so), but they don't work.
![Bildschirmfoto 2023-07-09 um 14 30 44](https://github.com/aralroca/next-translate/assets/80278171/9b90cac2-9593-437b-92c8-f97b51634fbf)

The translations exist:
![Bildschirmfoto 2023-07-09 um 14 31 54](https://github.com/aralroca/next-translate/assets/80278171/d9ba655a-027b-4752-a8ad-a096ecb49189)

Also, I get a hydration error when loading the page (but I don't know whether it's relevant. It just showed up since I use next translation. )

**Comment by aralroca** [comment]
@PixelPage-YT can you try in 2.5.0-canary.3? thanks

**Comment by PixelPage-YT** [comment]
> 2.5.0-canary.3

Sorry, I couldn't find a 2.5.0-canary.3
![Bildschirmfoto 2023-07-13 um 13 14 47](https://github.com/aralroca/next-translate/assets/80278171/f9583cbd-96d4-46b6-8a1b-99b3a5c977e6)

**Comment by aralroca** [comment]
next-translate-plugin

**Comment by PixelPage-YT** [comment]
> @PixelPage-YT next-translate-plugin

Ah, thanks. I tried it out, but nothing changed, except I don't get hydration errors any more.
Also, when I click on a <Link href="/content" />, it redirects to "/content" and not "/en/content"

**Comment by aralroca** [comment]
@PixelPage-YT would be very useful if you can share a reproducible repo example, thanks!

**Comment by PixelPage-YT** [comment]
> @PixelPage-YT would be very useful if you can share a reproducible repo example, thanks!

that's the problem. it works completely fine in other projects. i need to find the problem...
It might be useful to detect when the namespaces can't be loaded and log the error why they can't be.

**Comment by PixelPage-YT** [comment]
@aralroca Will you have a look over it?
Or do you know a workaround because i need it to work pretty soon...

**Comment by aralroca** [comment]
If I can't reproduce the error is hard for me to help you. Would be great if you can simplify and remove all the sensitive data from your project to share it 🙏

**Comment by PixelPage-YT** [comment]
I upgraded my deps again and somehow it works again. I hope it will continue to do so.
but i still need a way to redirect

> Also, when I click on a , it redirects to "/content" and not "/en/content"

**Comment by PixelPage-YT** [comment]
Update: just restarted my computer and it doesn't work again. I don't know what to do any more

**Comment by aralroca** [comment]
Please share your code, otherwise it is impossible for us to know what is wrong with you. Thank you

**Comment by aralroca** [comment]
My problem is still this: 
> When I click on a , it redirects to "/content" and not "/en/content"

This is because the i18n routing part from Next.js is only implemented in `pages` folder, in `app` folder you need to specify the locale in the href of the link.

**Comment by PixelPage-YT** [comment]
@aralroca 
thank you, do you have solutions for this?

> Also, how can I make /invalidlocale redirect to /404

And I added this to my config file: "/": ["index", "contact", "story"], The problem is that all the translations are loaded on the loading page, but for some reason the server always gets the namespaces from /. So it works now.

**Comment by aralroca** [comment]
> Also, how can I make /invalidlocale redirect to /404

Yes, here is an example using `redirect` from `next/navigation`:

https://github.com/aralroca/next-translate/blob/de5ad45d005488b36f21fed291e18437af468972/examples/with-app-directory/src/app/%5Blang%5D/layout.tsx#L12-L13
